### PR TITLE
Added Api contract test case for cdn_federation, user_federation and delivery_service_federation endpoint

### PR DESCRIPTION
**Pull request by gokulakrishnansvm**
Added Api contract test case for cdn_federation, user_federation and delivery_service_federation endpoint
<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops Test

## What is the best way to verify this PR?

Step 1 : Follow steps from README file(traffic_ops/testing/api_contract) to run api-contract-tests.

Step 2 : To execute API Contract Testcase , navigate to traffic_ops/testing/api_contract/v4
pytest --to-user Username --to-password Password --to-url URL --config Config Path --request-template Request template --response-template Response template 

Note: Config, Request template and response template paths are optional. 

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 1]
```diff
@@ -1757,5 +1757,95 @@
...
+            },
+            "ttl": {
+                "type": "integer"
+            },
+            "description": {
+                "type": "string"
+            },
+            "lastUpdated": {
+                "type": "string"
+            }
```
Comment (ericholguin) [inline traffic_ops/testing/api_contract/v4/data/response_template.json:1785]: are we missing some fields here? 
```
    {
        "id": 1,
        "cname": "test.quest.",
        "ttl": 48,
        "description": "A test federation",
        "lastUpdated": "2018-12-05 00:05:16+00",
        "deliveryService": {
            "id": 1,
            "xmlId": "demo1"
        }
    }
```

File: traffic_control/clients/python/trafficops/tosession.py [hunk 2]
```diff
@@ -1244,7 +1255,7 @@ def create_federation_in_cdn(self, cdn_name=None, data=None):
 		"""
 
 	@api_request('put', 'cdns/{cdn_name:s}/federations/{federation_id:d}', ('3.0', '4.0', '4.1', '5.0'))
-	def update_federation_in_cdn(self, cdn_name=None, federation_id=None, query_params=None):
+	def update_federation_in_cdn(self, cdn_name=None, federation_id=None, data=None):
```
[the remark this answers is not in the document]
Comment (gokulakrishnansvm) [reply traffic_control/clients/python/trafficops/tosession.py:1258]: For updating federation_in_cdn, We need to pass some data, Previously there was no argument for passing data, Instead query params was there, Here there is no use for Query params, So i removed and added data as an argument.

File: traffic_control/clients/python/trafficops/tosession.py [hunk 3]
```diff
@@ -1217,6 +1217,17 @@ def create_federation(self, data=None):
 		:raises: Union[LoginError, OperationError]
 		"""
 
+	@api_request('delete', 'federations', ('3.0', '4.0', '4.1', '5.0'))
+	def delete_federation(self):
+		"""
+		Allows a user to delete federations for their delivery service(s).
+		:ref:`to-api-federations`
+		:param data: The update action. QueueUpdateRequest() can be used for this argument also.
+		:type data: Dict[str, Any]
```
[the remark this answers is not in the document]
Comment (gokulakrishnansvm) [reply traffic_control/clients/python/trafficops/tosession.py:1226]: Removed it.

File: traffic_ops/testing/api_contract/v4/data/response_template.json (hunk 1 above)
Comment (gokulakrishnansvm) [reply traffic_ops/testing/api_contract/v4/data/response_template.json:1785]: Yes, I missed deliveryService , Added it.

**Review by ericholguin — APPROVED** [review]
LGTM

**Pull request merged**